Bài tập Shape-Triangle!

AlexWoo176 opened the pull request:
Gửi cô giáo bài tập ạ

AlexWoo176 committed:
Add files via upload

AlexWoo176 committed:
Exercise-Interface-HoangTuan

[Bài tập ] Triển khai interface 'Resizeable' cho các lớp hình học!

AlexWoo176 committed:
Management-Transactions

Hi, Ms Châu!
Xây dựng chương trình quản lý danh sách các giao dịch

quannguyen206 commented on src/ForeignExchangeTransactions.java:
```diff
@@ -0,0 +1,72 @@
+        this.exchangeRate = exchangeRate;
+    }
+
+    Scanner scanner = new Scanner(System.in);
+
+    public void inputData2() {
+        super.inputData();
+        System.out.println("Tỷ Giá Hối Đoái: ");
+        exchangeRate = scanner.nextFloat();
+        System.out.println("Loại Tiện Giao Dịch: ");
+        currency = scanner.nextInt();
+    }
```
xóa bỏ hàm này và đưa việc nhập giá tị vào trong hàm main

quannguyen206 commented on src/GoldTrading.java:
```diff
@@ -0,0 +1,46 @@
+        this.typeGold = typeGold;
+    }
+
+    Scanner scanner = new Scanner(System.in);
+    public void inputData1() {
+        super.inputData();
+        System.out.println("Loại Vàng Giao Dịch: ");
+        typeGold = scanner.next();
+    }
+
```
xóa bỏ hàm này và đưa vào trong main

quannguyen206 commented on src/ManageTransactions.java:
```diff
@@ -0,0 +1,38 @@
+
+        do {
+
+            System.out.println("\n\t\t------------------------------------");
+            System.out.println("\t\t|0. Thoat ung dung\t\t\t\t ");
+            System.out.println("\t\t|1. Them Giao Dich Vang\t\t\t\t ");
+            System.out.println("\t\t|2. Them Giao Dich Tien Te\t\t\t ");
+            System.out.println("\t\t|3. Hien Thi Danh Sach Giao Dich\t\t ");
+            System.out.println("\t\t|4. Xem Tong So Luong cua cac Giao Dich\t\t ");
+            System.out.println("\t\t--------------------------------------");
```
nên tách chỗ nãy ra thành 1 hàm menu() để gọi được nhiều lần và dễ sửa

quannguyen206 commented on src/Transaction.java:
```diff
@@ -0,0 +1,93 @@
+        this.totalAmount = totalAmount;
+    }
+
+    public void inputData() {
+        System.out.println("Mã Giao Dịch: ");
+        tradingCode = scanner.nextInt();
+        System.out.println("Ngày Giao Dịch: ");
+        dayTrading = scanner.next();
+        System.out.println("Đơn Giá: ");
+        unitPrice = scanner.nextDouble();
+        System.out.println("Số lượng: ");
+        quantities = scanner.nextInt();
+    }
```
xóa và đưa hàm này vào main

quan200699 commented on src/resizeable/shape/Circle.java:
```diff
@@ -0,0 +1,51 @@
+package resizeable.shape;
+
+public class Circle extends Shape implements Resizeable{
+    private double radius = 1.0;
```
khai báo giá trị mặc định nên để vào trong contructor

quan200699 commented on src/resizeable/shape/Circle.java:
```diff
@@ -0,0 +1,51 @@
+public class Circle extends Shape implements Resizeable{
+    private double radius = 1.0;
+
+    public Circle() {
```
this.radius = 1.0;

quan200699 commented on src/resizeable/shape/Rectangle.java:
```diff
@@ -0,0 +1,64 @@
+package resizeable.shape;
+
+public class Rectangle extends Shape implements Resizeable {
+    private double width = 1.0;
+    private double length = 1.0;
```
tương tự class trên

quan200699 commented on src/resizeable/shape/Shape.java:
```diff
@@ -0,0 +1,38 @@
+package resizeable.shape;
+
+public class Shape{
+    private String color = "Green";
+    private boolean filled = true;
```
tương tự trên

AlexWoo176 committed:
Create BinarySearchTree

AlexWoo176 committed:
Loi

AlexWoo176 committed:
Binary Search Tree

Gửi Ms Châu & Quân bài tập thao tác với Binary Search Tree.

AlexWoo176 committed:
Recursive-Binary-Search-Tree (

Gửi Ms.Châu & Quân Code "Bài tập Cài đặt thuật toán tìm kiếm nhị phân sử dụng đệ quy!"

AlexWoo176 committed:
Lỗi

AlexWoo176 committed:
Delete BinarySearchTree.java

AlexWoo176 committed:
Delete MainBST.java

AlexWoo176 committed:
Delete TreeNode.java

AlexWoo176 committed:
Delete BinarySearchTree.class

AlexWoo176 committed:
Delete MainBST.class

AlexWoo176 committed:
Delete TreeNode.class

AlexWoo176 committed:
Delete Delete-Binary-Search-Tree.iml

AlexWoo176 committed:
Thuật toán tìm kiếm

Gửi Ms.Châu & Quân!

AlexWoo176 committed:
Demo Illegal Triangle Exception

Gửi Ms Châu & Quân

AlexWoo176 committed:
Demo - Java - IO

Gửi Ms Châu & Quân, bài tập phần Java - I/O

AlexWoo176 committed:
[Bài tập] Quản lý sản phẩm lưu ra file nhị phân

Gửi Ms Châu & Quân. Mình gửi bài tập QLSP, bài tập này mình chưa hoàn thiện theo ý mong muốn. 
- Khi thêm sản phẩm mới việc lưu thông tin sản phẩm vào file chưa hoàn thành. (Chỉ hiện thị được trên màn hình kết quả)
Rất mong được sự góp ý của Quân & Ms.Châu để mình hoàn thiện bài này

quannguyen206 commented on src/repository/ProductRepo.java:
```diff
@@ -0,0 +1,32 @@
+    List<Product> showProductList();
+
+    @Override
+    void add(Product product);
+
+    @Override
+    void editProduct(int index, Product product);
```
nên để kiểu trả về là Product để có thể lấy ra giá trị thêm vào hoặc sau sửa

quannguyen206 commented on src/repository/ProductRepo.java:
```diff
@@ -0,0 +1,32 @@
+    void removeProduct(int index);
+
+    @Override
+    String findByName(String name);
```
hàm findByName nên trả về kiểu Product

quannguyen206 commented on src/repository/ProductRepo.java:
```diff
@@ -0,0 +1,32 @@
+    String findByName(String name);
+
+    @Override
+    void findByID(int id);
```
findById là hàm tìm kiếm nên sẽ phải có kiểu trả về là Product

quannguyen206 commented on src/views/MainTest.java:
```diff
@@ -0,0 +1,109 @@
+                case 3:
+                    System.out.println("Enter index for editing: ");
+                    int indexEdit = scanner.nextInt();
+                    productController.editProduct(indexEdit, createProduct());
+                    break;
+                case 4:
+                    System.out.println("Enter index for removing:");
+                    int indexRemove = scanner.nextInt();
+                    productController.removeProduct(indexRemove);
+                    break;
+                case 5:
+                    System.out.println("Enter a product Name:  ");
+                    String name= scanner.nextLine();
+                    System.out.println(productController.findByName(name));
+                    break;
+                case 6:
+                    System.out.println("Enter a product ID:  ");
+                    int id = scanner.nextInt();
+                    productController.findById(id);
+                    break;
+                case 7:
+                    System.out.println("Arrange product list upPrice");
+                    productController.sortUp();
+                    break;
+                case 8:
+                    System.out.println(" Arrange product list downPrice:");
+                    productController.sortDown();
+                    break;
+                case 9:
+                    System.out.println("Storing Product Information");
+                    ProductFile productFile = new ProductFile();
+                    productFile.writeFile();
+                    productFile.readFile();
+                    break;
+                case 0:
+                    System.exit(0);
+                default:
+                    System.out.println("You didn't make any choice");
+
+            }
```
có thể tách chỗ này ra thành 1 hàm selectMenu
public static void selectMenu(int choice)

quannguyen206 reviewed (commented):
Việc thêm sản phẩm mới cx như chỉnh sửa sản phẩm thì a sẽ lưu lại bằng cách là sẽ xóa toàn bộ dữ liệu file đang có và ghi lại list product vào trong file với dữ liệu mới được cập nhật
